momento-sdk/src/main/java/momento/sdk/CacheClient.java

```diff
@@ -85,6 +88,48 @@ public CacheClient(
     logger.debug("Control endpoint: " + credentialProvider.getControlEndpoint());
   }
 
+  /**
+   * Constructs a CacheClient.
+   *
+   * @param credentialProvider Provider for the credentials required to connect to Momento.
+   * @param configuration Configuration object containing all tunable client settings.
+   * @param itemDefaultTtl The default TTL for values written to a cache.
+   * @param eagerConnectionTimeout The timeout value while trying to establish an eager connection
+   *     with the Momento server.
+   * @return CacheClient
+   */
+  public static CacheClient create(
+      @Nonnull CredentialProvider credentialProvider,
+      @Nonnull Configuration configuration,
+      @Nonnull Duration itemDefaultTtl,
+      @Nullable Duration eagerConnectionTimeout) {
+    final CacheClient client =
+        CacheClient.builder(credentialProvider, configuration, itemDefaultTtl).build();
+
+    final long timeout;
+    if (eagerConnectionTimeout == null) {
+      timeout = DEFAULT_EAGER_CONNECTION_TIMEOUT_SECONDS;
+    } else {
+      timeout = eagerConnectionTimeout.getSeconds();
+    }
+
+    checkTimeoutValid(timeout);
+
+    // a client can explicitly set the timeout to be 0 in which case we don't want to establish an
+    // eager connection
+    if (timeout != 0) {
+      client.scsDataClient.connect(timeout);
+    }
+
+    return client;
+  }
+
+  private static void checkTimeoutValid(long timeout) {
+    if (timeout < 0) {
```

Comment:
I feel like a negative timeout can reasonably be treated as infinite, treated as 0, or throw. I'd lean slightly against throwing since the method can still return an unconnected client in the worst case. I think it's a minor thing overall though.

Reply:
I agree with you. The only reason I'll keep it as is for now is because we have done the same in 3 other SDKs.